ec2-spot-sagemaker-managed-spot-training/sagemaker-deepar.ipynb

```diff
@@ -0,0 +1,671 @@
+    "        num_samples -- integer, number of samples to compute at prediction time (default: 100)\n",
+    "        quantiles -- list of strings specifying the quantiles to compute (default: [\"0.1\", \"0.5\", \"0.9\"])\n",
+    "        \n",
+    "        Return value: list of `pandas.DataFrame` objects, each containing the predictions\n",
+    "        \"\"\"\n",
+    "        prediction_times = [x.index[-1]+1 for x in ts]\n",
+    "        req = self.__encode_request(ts, cat, encoding, num_samples, quantiles)\n",
+    "        res = super(DeepARPredictor, self).predict(req)\n",
+    "        return self.__decode_response(res, prediction_times, encoding)\n",
+    "    \n",
+    "    def __encode_request(self, ts, cat, encoding, num_samples, quantiles):\n",
+    "        instances = [series_to_obj(ts[k], cat[k] if cat else None) for k in range(len(ts))]\n",
+    "        configuration = {\"num_samples\": num_samples, \"output_types\": [\"quantiles\"], \"quantiles\": quantiles}\n",
+    "        http_request_data = {\"instances\": instances, \"configuration\": configuration}\n",
+    "        return json.dumps(http_request_data).encode(encoding)\n",
+    "    \n",
+    "    def __decode_response(self, response, prediction_times, encoding):\n",
+    "        response_data = json.loads(response.decode(encoding))\n",
+    "        list_of_df = []\n",
+    "        for k in range(len(prediction_times)):\n",
+    "            prediction_index = pd.DatetimeIndex(start=prediction_times[k], freq=self.freq, periods=self.prediction_length)\n",
+    "            list_of_df.append(pd.DataFrame(data=response_data['predictions'][k]['quantiles'], index=prediction_index))\n",
+    "        return list_of_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = DeepARPredictor(\n",
+    "    endpoint=endpoint_name,\n",
+    "    sagemaker_session=sagemaker_session,\n",
+    "    content_type=\"application/json\"\n",
+    ")\n",
+    "predictor.set_prediction_parameters(freq, prediction_length)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Make predictions and plot results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can use the previously created `predictor` object. For simplicity, we will predict only the first few time series used for training, and compare the results with the actual data we kept in the test set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "list_of_df = predictor.predict(time_series_training[:5])\n",
+    "actual_data = time_series[:5]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for k in range(len(list_of_df)):\n",
+    "    plt.figure(figsize=(12,6))\n",
```

Comment:
This is my OCD kicking in back again; This is the only plot call that uses figsize. It may worth adding a the same figsize parameters to previous plots in the playbook, so all charts have the same size and look the same.